### ISSUE TITLE
Playback control

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#305]
```
It would be really great to have some commands to control the playback using 
the generic next, previous, play/pause, shuffle, repeat would be sufficient.

About the use case: now that I can control my SMS remotely, I don't need my 
phone in my pocket when I'm on my PC. So I use my phone as a player and I like 
to plug it into my stereo, which is kind of not accessible to my PC. Then, at 
least skipping song sometimes is a must...

Thank you.
```

Original issue reported on code.google.com by `Valom...@gmail.com` on 12 Nov 2012 at 10:28

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 29 Jan 2013 at 10:52
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Commands next, previous, play/pause, stop has been implemented.
Shuffle and repeat are not doable because they are not exposed.

Volume management up/down/mute/unmute has been added too.

A new beta version of GTalkSMS will come soon.
```

Original comment by `Florent....@gmail.com` on 29 Jan 2013 at 10:16
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 16 Feb 2013 at 9:45
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
New version 4.4 Beta 1 is available in the download section
```

Original comment by `Florent....@gmail.com` on 16 Feb 2013 at 9:47

**Comment by GoogleCodeExporter** [comment]
```
Thank you so much for your work on this feature!
However, I haven't managed to make it work yet. Is there any restriction on the 
music player to use? On the Android version?

I've been trying to use it on Android 4.0, with the players Google Music and 
Huawei's Music+, to no effect. I don't get any message back from the smartphone 
(not even an error).

Please, let me now how if I can help.

Thanks again!
```

Original comment by `Valom...@gmail.com` on 17 Feb 2013 at 1:02

**Comment by GoogleCodeExporter** [comment]
```
Hi, 
This command works like a Bluetooth headset. It sends keycodes to control the 
music app. With some devices, you have to start your music application manually 
before.

If it's still not working can you provide me some debug logs plz ?

Thanks
```

Original comment by `Florent....@gmail.com` on 17 Feb 2013 at 3:50

**Comment by GoogleCodeExporter** [comment]
```
02-17 17:40:36.938 D/gtalksms( 4356): MainService executing command: 
"music:play"
02-17 17:40:36.938 D/gtalksms( 4356): Sending event key 126
02-17 17:40:36.951 E/gtalksms( 4356): Error sending event key 
dispatchMediaKeyEvent [class android.view.KeyEvent]
02-17 17:40:36.951 E/gtalksms( 4356): java.lang.NoSuchMethodException: 
dispatchMediaKeyEvent [class android.view.KeyEvent]
02-17 17:40:36.951 E/gtalksms( 4356): at 
java.lang.Class.getConstructorOrMethod(Class.java:460)
02-17 17:40:36.951 E/gtalksms( 4356): at 
java.lang.Class.getDeclaredMethod(Class.java:685)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.cmd.MusicCmd.handleMediaKeyEvent(MusicCmd.java:74)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.cmd.MusicCmd.sendKeyEvent(MusicCmd.java:59)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.cmd.MusicCmd.execute(MusicCmd.java:30)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.cmd.CommandHandlerBase.execute(CommandHandlerBase.java:2
59)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.cmd.CommandHandlerBase.execute(CommandHandlerBase.java:2
28)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.MainService.executeCommand(MainService.java:593)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.MainService.handleCommandFromXMPP(MainService.java:725)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.MainService.onHandleIntent(MainService.java:256)
02-17 17:40:36.951 E/gtalksms( 4356): at 
com.googlecode.gtalksms.MainService$ServiceHandler.handleMessage(MainService.jav
a:150)
02-17 17:40:36.951 E/gtalksms( 4356): at 
android.os.Handler.dispatchMessage(Handler.java:99)
02-17 17:40:36.951 E/gtalksms( 4356): at android.os.Looper.loop(Looper.java:137)
02-17 17:40:36.951 E/gtalksms( 4356): at 
android.os.HandlerThread.run(HandlerThread.java:60)

So, maybe this occurs because of a non-standard build by Huawei.
Perhaps this would do the trick: 
https://groups.google.com/forum/?fromgroups=#!topic/taskerdev/tClW-drrHwg

I'll probably try it when I'll have more time. Do you think that could be it?
```

Original comment by `Valom...@gmail.com` on 17 Feb 2013 at 5:05

**Comment by GoogleCodeExporter** [comment]
```
I'm using reflexion for this one because it was the only way too get it work on 
my phone GS2 with cyanogen.

So the broadcast solution might work on your phone.
Keep me posted if it works
```

Original comment by `Florent....@gmail.com` on 18 Feb 2013 at 10:02

**Comment by GoogleCodeExporter** [comment]
```
Well, it actually works! :)
I've attached the changes I made, according to the previously-mentioned 
https://groups.google.com/forum/?fromgroups=#!topic/taskerdev/tClW-drrHwg .
I'm not exactly sure about the use of sContext, I'm not familiar with android 
dev. Anyway, it works that way.

Thank you!

```

Original comment by `Valom...@gmail.com` on 3 Mar 2013 at 1:06

Attachments:
- [fix-event-handling-musiccmd.patch](https://storage.googleapis.com/google-code-attachments/gtalksms/issue-305/comment-9/fix-event-handling-musiccmd.patch)

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 7a72c73f42fc.
```

Original comment by `yako...@gmail.com` on 18 Mar 2013 at 7:47

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 4.4
```

Original comment by `Florent....@gmail.com` on 15 Sep 2013 at 3:27
- Changed state: **Deployed**
